sktime/annotation/ggs.py

```diff
@@ -0,0 +1,412 @@
+# -*- coding: utf-8 -*-
+"""
+Greedy Gaussian Segmentation (GGS).
+
+Based on:
+
+    Hallac, D., Nystrup, P. & Boyd, S.
+    Greedy Gaussian segmentation of multivariate time series.
+    Adv Data Anal Classif 13, 727–751 (2019).
+    https://doi.org/10.1007/s11634-018-0335-0
+
+- source code adapted based on: https://github.com/cvxgrp/GGS
+- paper available at: https://stanford.edu/~boyd/papers/pdf/ggs.pdf
+"""
+
+import logging
+import math
+from typing import Dict, List, Tuple
+
+import numpy as np
+import numpy.typing as npt
+from attrs import asdict, define, field
+from sklearn.utils.validation import check_random_state
+
+logger = logging.getLogger(__name__)
+
+
+@define
+class GGS:
+    """
+    Greedy Gaussian Segmentation.
+
+    Parameters
+    ----------
+        k_max: maximum number of change points to find
+        lamb: regularization parameter
+        max_shuffles: maximum number of shuffles
+        verbose: If ``True`` verbose output is enabled.
+        random_state: either random seed or an instance of ``np.random.RandomState``
+    """
+
+    k_max: int = 10
+    lamb: float = 1.0
+    max_shuffles: int = 250
+    verbose: bool = False
+    random_state: int = None
+
+    change_points_: npt.ArrayLike = field(init=False, default=[])
+    _intermediate_change_points: List[List[int]] = field(init=False, default=[])
+    _intermediate_ll: List[float] = field(init=False, default=[])
+
+    def initialize_intermediates(self):
+        """Fit."""
+        self._intermediate_change_points = []
+        self._intermediate_ll = []
+
+    def log_likelihood(self, cov: float, nrows: int, ncols: int) -> float:
+        """
+        Compute the GGS log likelihood.
+
```

Comment:
please state *which* log-lokelihood is being computed here.

Reply:
not sure what you mean, have you checked the paper?